community/kernel/src/main/java/org/neo4j/kernel/impl/storageengine/impl/recordstorage/RecordStorageEngine.java

```diff
@@ -207,10 +214,13 @@ public RecordStorageEngine(
             schemaCache = new SchemaCache( constraintSemantics, Collections.emptyList() );
             schemaStorage = new SchemaStorage( neoStores.getSchemaStore() );
 
-            labelScanStore = labelScanStoreProvider.getLabelScanStore();
+            NeoStoreIndexStoreView neoStoreIndexStoreView = new NeoStoreIndexStoreView( lockService, neoStores );
+            Boolean readOnly = config.get( GraphDatabaseSettings.read_only ) && operationalMode == OperationalMode.single;
```

Comment:
Yes, because in a cluster, all of the machines need to be `read_only` for it to be true and this is not verified in HA or causal clustering.
Problems will arise if a slave is configured to be `read_only` but master is not. The slave should thus still receive updates from master which violates the its `read_only` state.

Conclusion is that this setting only makes sense for single instance.